src/canopy/knowledge_base/reranker/cohere.py

```diff
@@ -0,0 +1,69 @@
+from typing import List, Optional
+
+from canopy.knowledge_base.models import KBQueryResult
+from canopy.knowledge_base.reranker import Reranker
+
+try:
+    import cohere
+except (OSError, ImportError, ModuleNotFoundError):
+    _cohere_installed = False
+else:
+    _cohere_installed = True
+
+
+class CohereReranker(Reranker):
+    """
+    Reranker that uses Cohere's text embedding to rerank documents.
+
+    For each query and documents returned for that query, returns a list
+    of documents ordered by their relevance to the provided query.
+    """
+
+    def __init__(self,
+                 model_name: str = 'rerank-english-v2.0',
+                 *,
+                 top_n: int = 10,
```

Comment:
Not a must but you can consider allowing pass `max_retries` and `timeout` and populate it to Cohere client

Reply:
Let's try to keep it simple for now, if we need it we can add it